src/runtimes/docker/utils/spawnStreamAsync.ts

```diff
@@ -229,7 +231,39 @@ export async function spawnStreamAsync(
     }
 
     if (options.onCommand) {
-        options.onCommand([command, ...args].join(' '));
+        options.onCommand([command, ...normalizedArgs].join(' '));
+    }
+
+    ext.outputChannel.debug([command, ...normalizedArgs].join(' '));
+
+    let stdOutPipe: NodeJS.WritableStream | undefined = options.stdOutPipe;
```

Comment:
I don't know that it makes sense to do a diagnostics only stream runner, since things like error messages we potentially want to log (as an error) no matter what. Some of the diagnostics can be split out into methods to keep the main code cleaner, but I think it makes sense to have debug/diagnostics somewhat inline with the code being logged. My main reason for doing this in spawnStreamAsync was that it's the lowest level in how we spawn processes, since we use it in the command runners as well as execAsync. Doing the logging here ensures that it applies to all the commands we run.

Reply:
Having it here is going to be problematic for refactoring to use this library as a package, as well.

Reply:
Very good point.